### PR TITLE
initial version of the 2024 christmas article - Dec 24

### PR DESCRIPTION
**Pull request by 2shortplanks**

### CODE REVIEW

File: 2024/articles/2024-12-25.pod [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+I started taking submissions.  I eventually gave the calendar away for others
+to maintain, took it back again years later, and then finally gave it
+away again.  Like any good project, it's totally grown beyond me and has a life
+of its own.
+
+The crucial thing about the calendar - more important than any one article,
+any one year, or event any one programming language - is the idea that we can
+all enjoy learning about our favorite programming language.  Things don't have
+to be taught in boring instructions, but the instead having something to learn
+about can be silly, daft, fun, and a treat!  Why not have a story about how
```
Comment (PatrickCronin) [inline 2024/articles/2024-12-25.pod:257]: Doesn't read well. Maybe something like "but the instead having something to learn about" => "but instead learning about something"?

File: 2024/articles/2024-12-25.pod [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+Santa's elves are all Perl programmers faced with saving Christmas, and the
+only way to do it is to use this module?  Why not have a tale about how we can
+control Christmas lights with Perl?  Create Christmas memes?  Many a clever
+person in the Perl community has talked about optomizing projects for fun -
```
Comment (PatrickCronin) [inline 2024/articles/2024-12-25.pod:261]: optomizing => optimizing

File: 2024/articles/2024-12-25.pod [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+attracting volunteers by making projects enjoyable to be a part of - and the
+Perl Advent Calendar embraces this idea of rewarding everyone that reads it
+with joy.  This popular format has been copied by other languages.
+
+=head2 A Merry Christmas to us all; Bless us, everyone
+
+So, as I try to do every year, and on this twenty fifth year of the Perl Advent
```
Comment (PatrickCronin) [inline 2024/articles/2024-12-25.pod:268]: twenty fifth should be hyphenated

**Review by PatrickCronin — CHANGES_REQUESTED** [review]
What a treat. A heartfelt summary of what Perl and the calendar has brought the world over 25 years. The only comments I have are grammatical / spelling.

**Comment by oalders** [comment]
Thank you for your contribution! I'm finding that blog articles which include images have much more compelling social media previews when the article is shared. Adding an image is by no means required and you should feel free to ignore this message entirely. However, if you'd like to add an image, it's as simple as:

* Adding the image to the `2024/share/static` folder like `2024/share/static/some-graphic.png`
* Including the image as HTML in your article

The HTML would look like:

```
=for :html
<img src="some-graphic.png">

```

The trailing blank line is important as that signals that the `HTML` section is finished.

Images can most certainly be AI generated. You should also feel free to use https://search.creativecommons.org/ in order to find CC images which can be freely shared. If you choose to go this route, please provide proper attribution when including the image, if attribution is required.

Most importantly, if you don't feel like doing this, then feel free to submit your article without an image. It's *your* article and *you* need to be happy with it.

**Comment by oalders** [comment]
@2shortplanks do you know when you might be able to have a look at the review comments?

File: 2024/articles/2024-12-25.pod [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+that we never would have imagined twenty five years ago.  By making installing
+software easier we freed ourselves from the concept of the "core" Perl language
+extending what we could use out into reliably having whatever bits of CPAN we
+want to hand.  This brought the philosophy of "CPAN is the language" to the
+forefront.
+
+Much of the work on Perl in the last twenty five years has been adapting the
+core interpreter so that it's possible to modify it more and more by simply
+installing a module.  Where we once begrudgingly replied on source code
+modifications to manipulate the code we write to add brand new language features
```
[the remark this answers is not in the document]
Comment (2shortplanks) [reply 2024/articles/2024-12-25.pod:54]: Rewrote to not suck.

**Comment by 2shortplanks** [comment]
@oalders Image added as requested.

You changed the title of the pull request here to suggest this article is for the 24th.  Just to remind you, the pull request currently puts it at the 25th - if you want to change that you'll need to modify things!

**Review by PatrickCronin — APPROVED** [review]
Looks great! Thanks!

**Comment by oalders** [comment]
Thanks @PatrickCronin for the review and @2shortplanks for taking the time to write this article. I think this is a lovely way to finish up year 25. ❤️

**Pull request merged**